articles/tenntenn.re

```diff
@@ -0,0 +1,758 @@
+= Goの新しいコントラクト
+
+== 始めに
+
+株式会社メルペイのバックエンドエンジニアの@tenntenn@<fn>{tenntenn}です。
+本稿はGopherCon 2019のIan Lance Taylor氏の"Generics in Go"@<fn>{goingenerics-liveblog}というセッションとその後公開されたDesign Doc@<fn>{contract-draft2}を元に書いています。サンプルコードなどは一部を除き、Design Docから引用しています。
+
+//footnote[tenntenn][@<href>{https://tenntenn.dev}]
+//footnote[goingenerics-liveblog][@<href>{https://about.sourcegraph.com/go/gophercon-2019-generics-in-go}]
+//footnote[contract-draft2][@<href>{https://go.googlesource.com/proposal/+/master/design/go2draft-contracts.md}]
+
+なお、2019年8月時点では本稿で扱う内容がそのままGo2に入るという話ではありません。
+「Go2でジェネリクスが入る」という情報は正確ではなく、新しい機能提案の1つとしてジェネリクスが議論されているというだけです。
+むしろ、後方互換性がうまく保たれ、よい実装が登場すればGo1.Xでリリースされることすらあるでしょう。
+
+本稿ではGoにおけるジェネリクスの必要性については議論しません。
+その点について知りたい場合は、Goの公式ブログの記事"Why Generics?"@<fn>{why-generics}を参照するとよいでしょう。
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Disallow to use "?". (ja-technical-writing/no-exclamation-question-mark)